apps/documents/serializers.py

```diff
@@ -94,3 +94,15 @@ def get_presigned_s3(self, obj):
         log_metric('transmission.info', tags={'method': 'documents.generate_presigned_url', 'module': __name__})
 
         return url
+
+
+class DocumentUpdateSerializer(DocumentRetrieveSerializer):
+
+    class Meta:
+        model = Document
+        exclude = ('shipment',)
+        if settings.PROFILES_ENABLED:
+            read_only_fields = ('owner_id', 'document_type', 'file_type', 'name', 'description')
```

Comment:
I think it should be okay for users to update name/description.
```suggestion
            read_only_fields = ('owner_id', 'document_type', 'file_type')
```